sfkit/sidecar/server.py

```diff
@@ -0,0 +1,73 @@
+import json
+import os
+import select
+import socket
+import subprocess
+from threading import Thread
+
+from sfkit.sidecar.utils import get_sock_path
+
+
+def handle_client(client: socket.socket):
+    try:
+        while True:
+            data = client.recv(1024)
+            if not data:
+                break
+            try:
+                request = json.loads(data.decode("utf-8"))
+            except json.JSONDecodeError:
+                client.sendall("Invalid JSON format".encode("utf-8"))
+                continue
+
+            study_id = request.get("study_id", "")
+            data_path = request.get("data_path", "")
```

Comment:
```suggestion
            data_path = os.path.realpath(request.get("data_path", ""))
            
            if not data_path.startswith(constants.SAFE_DATA_PATH):
                client.sendall("Invalid data_path".encode("utf-8"))
                client.close()
                return
```
This avoids a possible path traversal attack.. We should also pair this with
```py
# constants.py

SAFE_DATA_PATH = os.environ.get("SAFE_DATA_PATH", "/data")
SAFE_DATA_PATH = os.path.join(os.path.realpath(SAFE_DATA_PATH), "")
```
(`SAFE_DATA_PATH` env var will be set to `/home/jupyter` when running as a Leo VM container)

`realpath` here resolves any symbolic links (if present), and `join` with `""` ensures there's a trailing slash, which is important to reject user-defined paths like `/home/jupyter2` later.